### PR TITLE
Change: Make all dropdown lists extend width if necessary.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

Dropdown list items can sometimes be too long for the parent widget. In most cases we try to make the parent widget wide enough, but in some cases this becomes too wide.

image

## Description

This PR removes the default-false auto_width parameter, and makes all dropdowns automatically extend in width as required.

image

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/company_gui.cpp [hunk 1]
```diff
@@ -592,9 +592,14 @@ class DropDownListColourItem : public DropDownListItem {
...
 
+	uint Width() const override
+	{
+		return ScaleGUITrad(28) + WidgetDimensions::scaled.hsep_normal + GetStringBoundingBox(this->String()).width + WidgetDimensions::scaled.dropdowntext.Horizontal();
+	}
+
 	uint Height(uint width) const override
 	{
-		return std::max(FONT_HEIGHT_NORMAL, ScaleGUITrad(12) + 2);
+		return std::max(FONT_HEIGHT_NORMAL, ScaleGUITrad(12) + WidgetDimensions::scaled.vsep_normal);
```
Comment (2TallTyler) [inline src/company_gui.cpp:602]: Fixing scaling here doesn't really fit the commit name, could be its own commit or just a different commit title. (Unless you're planning to just squash the whole PR into one commit, of course.)

**Review by 2TallTyler — COMMENTED** [review]
Obviously we have to draw a line somewhere between strings that fit and strings that don't, but perhaps all the built-in options in English (UK) should fit?

nanes

**Comment by PeterN** [comment]
Yes we could do that, but this PR no longer touches the width of the dropdown widgets, only the width of the opened dropdowns.

There are other issues with dropdown widgets, the widget system resizes based on the current value when the window is opened and requires every window to manually work out what size is really required. I think these are another PR.

File: src/company_gui.cpp (hunk 1 above)
Comment (PeterN) [reply src/company_gui.cpp:602]: Updated wording on the commit.

**Review by 2TallTyler — APPROVED** [review]
